packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -64,6 +66,26 @@ enum _TextSelectionHandlePosition { start, end }
 /// Used by [TextSelectionOverlay.onSelectionOverlayChanged].
 typedef TextSelectionOverlayChanged = void Function(TextEditingValue value, Rect caretRect);
 
+/// Signature for when a pointer has started dragging to select text.
+///
+/// The [details] argument is the same as the one passed to [DragGestureRecognizer.onStart].
+typedef DragSelectionStartCallback = void Function(DragStartDetails details);
```

Comment:
Be consistent with the rest of the file (i.e. we didn't redefine the other various Gesture*Callback in this file). Is this needed?

Reply:
I did this because I needed to redefine the update callback to take two arguments. Do you think I should remove the start and end callback definitions and keep the update one?

Reply:
Yes, add a comment for the update one to explain how it's different from the rest.

Reply:
Never mind, addressed in the next comment